ps.go

```diff
@@ -73,28 +82,45 @@ var psCommand = cli.Command{
 				continue
 			}
 			fields := strings.Fields(line)
-			p, err := strconv.Atoi(fields[pidIndex])
-			if err != nil {
-				return fmt.Errorf("unexpected pid '%s': %s", fields[pidIndex], err)
-			}
-
-			for _, pid := range pids {
-				if pid == p {
-					fmt.Println(line)
-					break
-				}
+			if fields[pidIndex] != cmdPs {
+				fmt.Println(line)
 			}
 		}
 		return nil
-	},
+	}
+
+	ns := fmt.Sprintf("/proc/%d/ns/pid", pids[0])
+	fd, err := syscall.Open(ns, syscall.O_RDONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("open /proc/%d/ns/pid %v", pids[0], err)
+	}
+	defer syscall.Close(fd)
+	err = system.Setns(uintptr(fd), 0)
```

Comment:
The "send PIDs over an AF_UNIX socket" means you'll get PIDs in your current PID namespace. This was one of the extensions to the current set up I was considering (it means you could use `libcontainer.Container.Processes()` inside a rootless container -- since we don't have cgroups).